testing/librsynctest.py

```diff
@@ -32,8 +32,12 @@ def sig_file_test_helper(self, blocksize, iterations, file_len = None):
 		for i in range(iterations):
 			MakeRandomFile(self.basis.path, file_len)
 			self._clean_file(self.sig)
-			assert not os.system(b"rdiff -b %i -H md4 signature %s %s" %
-								 (blocksize, self.basis.path, self.sig.path))
+			if b'-H' in subprocess.check_output(["rdiff", "--help"]):
```

Comment:
Needs to be bytes (I assume). Python will not complain if you search for bytes in strings but it will also never find it.

Reply:
Why would `["rdiff", "--help"]` need to be bytes? 

(check_output returns bytes if you do not specify an encoding, see https://docs.python.org/3/library/subprocess.html#subprocess.check_output and https://docs.python.org/3/library/subprocess.html#subprocess.run (for stdout))

From my point of view (and my tests) everything that needs to be bytes (`b'-H'`) is already correct.

Maybe i missinterpreted your comment, could you elaborate what exactly in the statement you would like to see as bytes?

Reply:
@ericzolf `subprocess.check_output(["rdiff", "--help"])` is returning bytes. Not str.
https://docs.python.org/3/library/subprocess.html#subprocess.check_output

Current implementation is ok

Reply:
Ah, sorry, all over the place it's "bytes in, bytes out/str in, str out" but not here, you're right.